executor/insert.go

```diff
@@ -215,15 +215,22 @@ func (e *InsertExec) doDupRowUpdate(handle int64, oldRow []types.Datum, newRow [
 }
 
 // updateDupKeyValues updates the dupKeyValues for further duplicate key check.
-func (e *InsertExec) updateDupKeyValues(row toBeCheckedRow, oldHandle int64,
-	newHandle int64, handleChanged bool, updatedRow []types.Datum) error {
+func (e *InsertExec) updateDupKeyValues(oldHandle int64, newHandle int64,
+	handleChanged bool, oldRow []types.Datum, updatedRow []types.Datum) error {
 	// There is only one row per update.
 	fillBackKeysInRows, err := e.getKeysNeedCheck(e.ctx, e.Table, [][]types.Datum{updatedRow})
 	if err != nil {
 		return errors.Trace(err)
 	}
 	// Delete old keys and fill back new key-values of the updated row.
-	e.deleteDupKeys(row)
+	cleanupRows, err := e.getKeysNeedCheck(e.ctx, e.Table, [][]types.Datum{oldRow})
```

Comment:
We could use a function to handle this part and https://github.com/pingcap/tidb/blob/9070bb6b86b27e59f35ebc2f984e369d3c5a115d/executor/replace.go#L73
together.

Reply:
Thanks for the review @jackysp, I plan to define a function for this and put it in the base struct `InsertValues`, what do u think?

Reply:
I think to put it into `batchChecker` is better. Thanks for your contribution, @bb7133 !